src/Common/src/Interop/Windows/HttpApi/Interop.HttpApi.cs

```diff
@@ -0,0 +1,1017 @@
+                                if (bytesToRead > (uint)size)
+                                {
+                                    bytesToRead = (uint)size;
+                                }
+                                for (uint i = 0; i < bytesToRead; i++)
+                                {
+                                    *(pTo++) = *(pFrom++);
+                                }
+                                dataRead += bytesToRead;
+                                dataChunkOffset += bytesToRead;
+                            }
+                        }
+                    }
+                }
+                //we're finished.
+                if (dataChunkIndex == request->EntityChunkCount)
+                {
+                    dataChunkIndex = -1;
+                }
+            }
+
+            if (NetEventSource.IsEnabled)
+            {
+                NetEventSource.Exit(null);
+            }
+            return dataRead;
+        }
+
+        internal static HTTP_VERB GetKnownVerb(byte[] memoryBlob, IntPtr originalAddress)
+        {
+            NetEventSource.Enter(null);
+
+            // Return value.
+            HTTP_VERB verb = HTTP_VERB.HttpVerbUnknown;
+            fixed (byte* pMemoryBlob = memoryBlob)
+            {
+                HTTP_REQUEST* request = (HTTP_REQUEST*)pMemoryBlob;
+                if ((int)request->Verb > (int)HTTP_VERB.HttpVerbUnparsed && (int)request->Verb < (int)HTTP_VERB.HttpVerbMaximum)
+                {
+                    verb = request->Verb;
+                }
+            }
+
+            NetEventSource.Exit(null);
+            return verb;
+        }
+
+        internal static IPEndPoint GetRemoteEndPoint(byte[] memoryBlob, IntPtr originalAddress)
+        {
+            throw new NotImplementedException();
+            //NetEventSource.Enter(null);
+
+            //SocketAddress v4address = new SocketAddress(AddressFamily.InterNetwork, SocketAddress.IPv4AddressSize);
+            //SocketAddress v6address = new SocketAddress(AddressFamily.InterNetworkV6, SocketAddress.IPv6AddressSize);
+
+            //fixed (byte* pMemoryBlob = memoryBlob)
+            //{
+            //    HTTP_REQUEST* request = (HTTP_REQUEST*)pMemoryBlob;
+            //    IntPtr address = request->Address.pRemoteAddress != null ? (IntPtr)(pMemoryBlob - (byte*)originalAddress + (byte*)request->Address.pRemoteAddress) : IntPtr.Zero;
+            //    CopyOutAddress(address, ref v4address, ref v6address);
+            //}
+
+            //IPEndPoint endpoint = null;
+            //if (v4address != null)
+            //{
+            //    endpoint = IPEndPoint.Any.Create(v4address) as IPEndPoint;
+            //}
+            //else if (v6address != null)
+            //{
+            //    endpoint = IPEndPoint.IPv6Any.Create(v6address) as IPEndPoint;
+            //}
+
+            //NetEventSource.Exit(null);
+            //return endpoint;
```

Comment:
What's the plan for all of this commented out code?  Is there a TODO to follow-up on this?